src/misc.ts

```diff
@@ -0,0 +1,94 @@
+//
+// Types
+//
+
+/**
+ * Makes every specified property of the specified object type mutable.
+ *
+ * @template ObjectValue - The object whose readonly properties to make mutable.
+ * @template TargetKey - The property key(s) to make mutable.
+ */
+export type Mutable<
+  ObjectValue extends Record<string, unknown>,
+  TargetKey extends keyof ObjectValue
+> = {
+  -readonly [Key in keyof Pick<ObjectValue, TargetKey>]: ObjectValue[Key];
+} &
+  {
+    [Key in keyof Omit<ObjectValue, TargetKey>]: ObjectValue[Key];
+  };
+
+/**
+ * Useful for representing some value that _might_ be present and / or complete.
+ *
+ * @template Value - The value that might be present or complete.
+ */
+export type Maybe<Value> = Partial<Value> | null | undefined;
+
+/**
+ * Like {@link Array}, but always non-empty.
+ *
+ * @template Element - The non-empty array member type.
+ */
+export type NonEmptyArray<Element> = [Element, ...Element[]];
+
+/**
+ * A JavaScript object that is not `null`, a function, or an array. The object
+ * can still be an instance of a class.
+ */
+export type RuntimeObject = Record<number | string | symbol, unknown>;
```

Comment:
Did we extract this from an existing project or is this a new type we're introducing here? If the latter, then perhaps this should be `PlainObject` instead of `RuntimeObject`? It's not clear to me what the "runtime" part means (unless you mean something that can change at runtime, which covers arrays too).

Reply:
Ahh, right. Hmm, I'm struggling with the name too. Okay, this works then!